ChatApplication/Services/CacheExtensionsService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Abp.Json;
+using ChatApplication.Data;
+using ChatApplication.Models;
+using ChatApplication.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Distributed;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ChatApplication.Services
+{
+    public class CacheExtensionsService : ICacheExtensionsService
+    {
+        IDistributedCache _distributedCache;
+
+        public CacheExtensionsService(IDistributedCache distributedCache)
+        {
+            _distributedCache = distributedCache;
+        }
+
+        public async Task<IEnumerable<Message>> GetCacheValueAsync( string key)
+        {
+            var result = await _distributedCache.GetStringAsync(key);
+
+            if (String.IsNullOrEmpty(result))
+            {
+                return null;
```

Comment:
what if the cache TTL has passed? we should have fallback function here to set the cache again. Usually the TTL is 5 mins so the caches are expired after 5 minutes. See https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.distributed.distributedcacheentryoptions.absoluteexpiration?view=dotnet-plat-ext-6.0#microsoft-extensions-caching-distributed-distributedcacheentryoptions-absoluteexpiration